lib/core/src/Cardano/Wallet.hs

```diff
@@ -2142,9 +2143,13 @@ mkTxMetaWithoutSel blockHeader wState txCtx inps outs =
                 w@WithdrawalSelf{} -> Coin.add (withdrawalToCoin w)
                 WithdrawalExternal{} -> Prelude.id
                 NoWithdrawal -> Prelude.id
+        dir = if (txCtx ^. #txDelegationAction) == Just Quit then
+                   Incoming
+              else
+                   Outgoing
```

Comment:
Fair, the old tx workflow should also be able to produce pending incoming txs through [this logic](https://github.com/input-output-hk/cardano-wallet/blob/13a6c0b2c865092011ca0bc8c5ee2e7d7194365b/lib/core/src/Cardano/Wallet.hs#L2202).

I still suspect this could be a problem, but unrelated to this PR.